Add namespace for Concordium (ccd) (#83)
ChainAgnostic/namespaces: ccd/caip2.md

---
namespace-identifier: ccd-caip2
title: Concordium - Networks
author: "Concordium development team <developers@concordium.com>"
discussions-to: https://github.com/ChainAgnostic/namespaces/pull/83
status: Draft
type: Standard
created: 2023-04-01
---

<!--You can leave these HTML comments in your merged EIP and delete the 
 visible duplicate text guides, they will not appear and may be helpful to 
 refer to if you edit it again. This is the suggested template for new EIPs.
 Note that an EIP number will be assigned by an editor. When opening a pull
 request to submit your EIP, please use an abbreviated title in the 
 filename, `caipX.md`, all lowercase, no `-` between the CAIP and its 
 number.-->

> This is the suggested template for new CAIPs.

Reviewer: ```suggestion
```

# CAIP-2

*For context, see the [CAIP-2][] specification.*

<!-- Provide a simplified and layman-accessible explanation of the CAIP.-->

This CAIP defines an identification schema for Concordium networks.

> The Concordium networks are identified by `ccd` namespace followed by the hash of the network's genesis block.

Reviewer: ```suggestion
The Concordium networks are identified by the `ccd` namespace prefix, followed by a truncated hash of the given network's genesis block.
```

## Rationale
<!--A short (~200 word) description of the technical issue being addressed.-->
Concordium has one mainnet and potentially several testnets.
These networks are uniquely identified by a hash of their genesis blocks.

## Syntax

<!-- Explain the actual algorithm or transformation needed to transform inputs into a
conformant and unique CAIP deterministically.  Consider including a regular
expression for validation as well, as some consumers or toolmakers may want to
support this CAIP without a deep understanding of any specifications, devdocs,
or improvement proposals on which this specification depends. -->

> Concordium's network ID is a hash of its genesis block in the `hex` encoding truncated to the first 32 characters.

Reviewer: ```suggestion
Concordium's network ID is a hash of its genesis block in hexadecimal encoding, truncated to the first 32 characters.
```

<!-- TODO: add some details, e.g. something like pseudocode `truncate(SHA512(block_data), 32)` -->

### Resolution Mechanics

A hash of the genesis block can be obtained using the gRPC interface of a Concordium node (see [Concordium gRPC][]).

> The [GetConsensusInfo][] command provides data for constructing a network identifier.

Reviewer: ```suggestion
The [GetConsensusInfo][] command provides data which includes the hash of that network's genesis block, which can be truncated to serve as the network identifier.
```
